providers/gov/nih/normalizer.py

```diff
@@ -0,0 +1,166 @@
+from share.normalize import *
+from share.normalize.utils import format_address
+
+
+class Person(Parser):
+    given_name = ParseName(ctx).first
+    family_name = ParseName(ctx).last
+    additional_name = ParseName(ctx).middle
+    suffix = ParseName(ctx).suffix
+
+
+class Contributor(Parser):
+    order_cited = ctx('index')
+    cited_name = ctx.PI_NAME
+    person = Delegate(Person, ctx.PI_NAME)
+
+    class Extra:
+        pi_id = ctx.PI_ID
+
+
+class ProgramOfficer(Parser):
+    schema = 'Contributor'
+
+    order_cited = ctx('index')
+    cited_name = ctx
+    person = Delegate(Person, ctx)
+
+
+class Link(Parser):
+    url = ctx
+    type = Static('provider')
+
+
+class ThroughLinks(Parser):
+    link = Delegate(Link, ctx)
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = Delegate(Tag, ctx)
+
+
+class Funder(Parser):
+    # The full name of the IC, as defined here: http://grants.nih.gov/grants/glossary.htm#InstituteorCenter(IC)
+    name = ctx.IC_NAME
+
+    # The organizational code of the IC, as defined here: http://grants.nih.gov/grants/glossary.htm#InstituteorCenter(IC)
+    community_identifier = ctx.ADMINISTERING_IC
+
+    class Extra:
+        funding_ics = ctx.FUNDING_ICs
+        funding_mechanism = ctx.FUNDING_MECHANISM
+
+
+class ThroughAwardEntities(Parser):
+    entity = Delegate(Funder, ctx)
+
+
+class Award(Parser):
+
+    def get_award_amount(self, award_info):
+        if award_info:
+            return award_info.split(':')[1].replace('\\', '')
+        return None
+
+    # The amount of the award provided by the funding NIH Institute(s) or Center(s)
+    description = RunPython('get_award_amount', ctx.FUNDING_ICs)
+    entities = Map(Delegate(ThroughAwardEntities), ctx)
+
+    class Extra:
+        arra_funded = ctx.ARRA_FUNDED
+        award_notice_date = ctx.AWARD_NOTICE_DATE
+
+
+class ThroughAwards(Parser):
+    award = Delegate(Award, ctx)
+
+
+class Organization(Parser):
+
+    def format_address(self, doc):
+        return format_address(
+            self,
+            city=doc.get('ORG_CITY'),
+            state_or_province=doc.get('ORG_STATE'),
+            postal_code=doc.get('ORG_ZIPCODE'),
+            country=doc.get('ORG_COUNTRY')
+        )
+
+    name = ctx.ORG_NAME
+    location = RunPython('format_address', ctx)
+
+    class Extra:
+        organization_duns = ctx.ORG_DUNS
+        organization_fips = ctx.ORG_FIPS
+        organization_dept = ctx.ORG_DEPT
+        organization_district = ctx.ORG_DISTRICT
+
+
+class Association(Parser):
+    pass
+
+
+class Project(Parser):
+
+    PROJECT_BASE_URL = 'https://projectreporter.nih.gov/project_info_description.cfm?aid={}'
+    FOA_BASE_URL = 'https://grants.nih.gov/grants/guide/pa-files/{}.html'
+
+    def format_nih_url(self, id):
```

Comment:
https://github.com/CenterForOpenScience/SHARE/pull/218#discussion_r71533780

Note: the `Extra` class is also considered a class level variable but should go after other class level variables.